scripts/quests/windurst/To_Catch_A_Falling_Star.lua

```diff
@@ -0,0 +1,154 @@
+    item = xi.items.FISH_SCALE_SHIELD
+}
+
+quest.sections =
+{
+    -- Quest Acceptance
+    {
+        check = function(player, status, vars)
+           return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Sigismund'] = quest:progressEvent(196, 0, xi.items.STARFALL_TEAR),
+
+            onEventFinish =
+            {
+                [196] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- Supporting Character dialogue after Accepted
+    {
+        check  = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tohopka'] = quest:progressEvent(198, 0, xi.items.STARFALL_TEAR, xi.items.HANDFUL_OF_PUGIL_SCALES)
+        }
+    },
+
+    -- Twinkle Tree in West Saru.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WEST_SARUTABARUTA] =
+        {
+            ['Twinkle_Tree'] =
+            {
+                onTrigger = function(player, npc)
+                    if VanadielHour() <= 3 then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        return quest:noAction()
+                    end
+                    player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                    return quest:noAction()
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if VanadielHour() > 3 then
+                        player:messageSpecial(westSaruMessages.NOTHING_OUT_OF_ORDINARY)
+                        return quest:noAction()
+                    end
+
+                    if npcUtil.tradeHas(trade, xi.items.HANDFUL_OF_PUGIL_SCALES) then
+                        player:messageSpecial(westSaruMessages.FROST_DEPOSIT_TWINKLES)
+                        player:messageSpecial(westSaruMessages.MELT_BARE_HANDS)
+                        if npcUtil.giveItem(player, xi.items.STARFALL_TEAR) then
+                            player:confirmTrade()
+                            if quest:getVar(player, 'Prog') == 0 then
+                                quest:setVar(player, 'Prog', 1)
+                            end
+                        end
+                    end
+                    return quest:noAction()
+                end
+            }
+        }
+    },
+
+    -- Quest Complete
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Sigismund'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(197, 0, xi.items.STARFALL_TEAR)
```

Comment:
So because event 197 doesn't make any progress through the quest we can use the following to prevent this NPC from barring progress with any other quests.
```suggestion
                    return quest:event(197, 0, xi.items.STARFALL_TEAR)
```
By using event, it allows the NPC to cycle through it's available dialogues. progressEvent takes priority over all over dialogues the NPC is involved with.